server/src/main/java/org/apache/druid/metadata/SQLMetadataSupervisorManager.java

```diff
@@ -259,7 +261,8 @@ public Pair<String, SupervisorSpec> map(int index, ResultSet r, StatementContext
                           );
                         }
                         catch (IOException e) {
-                          throw new RuntimeException(e);
+                          log.error(e, "Could not map json payload to a SupervisorSpec for spec_id: [%s]", r.getString("spec_id"));
```

Comment:
Since this would be a customer facing exception, I had a few questions:  
* Will the bad supervisor be shown in the ingestion tab
* If the bad supervisor is shown in the status tab, what would be the status of the supervisor. 
* If the bad supervisor is not shown in the status tab, could you please make this error message more actionable. May be something like `Could not map json payload to a SupervisorSpec for spec_id: [%s]. Please remove the supervisor spec_id: [%s] from the database and then submit a new request via the overlord.`

Reply:
The bad supervisor doesn't show up in the status tab. I have updated the exception message.

Reply:
At present, this supervisor will not be removed within `KillSupervisors` since the `getLatest` method won't return it. 

I am wondering whether the removal of this supervisor from db should be handled within `KillSupervisors` coordinator duty or if it should be left for the users to manage?

Reply:
I think for the fist cut, let users manage it themselves but yeah removing it using killSupervisors seems like a better idea. 
That can be a different PR though.